### PR TITLE
INTMDB-387: Enable Azure NVME for Atlas Dedicated clusters

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

INTMDB-387: Enable Azure NVME for Atlas Dedicated clusters

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Review by themantissa — APPROVED** [review]
LGTM but fyi for @shum we didn't actually include the same in https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_advanced_cluster.go so either should work, and advanced is preferred, for those just starting out (we are going to have to move everyone from one to the other).   Still good to do till we deprecate clusters.

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: mongodbatlas/resource_mongodbatlas_cluster.go [hunk 1]
```diff
@@ -1148,7 +1148,7 @@ func expandProviderSetting(d *schema.ResourceData) (*matlas.ProviderSettings, er
 		providerSettings.AutoScaling = &matlas.AutoScaling{Compute: compute}
 	}
 
-	if d.Get("provider_name") == "AWS" {
+	if d.Get("provider_name") == "AWS" || d.Get("provider_name") == "AZURE" {
```
Comment (shum) [inline mongodbatlas/resource_mongodbatlas_cluster.go]: I just realized that `providerSettings.diskIOPS` is the Disk IOPS setting for AWS storage only. 

https://www.mongodb.com/docs/atlas/reference/api/clusters-modify-one/

Instead, we probably need to do the same with `providerSettings.diskTypeName` https://www.mongodb.com/docs/atlas/reference/api/clusters-modify-one/#std-label-modify-cluster-providerSettings-diskTypeName where the user cannot provide diskTypeName if the isntance size contains NVME.

something like:
```
if d.Get("provider_name") == "AZURE" {
		if v, ok := d.GetOk("provider_disk_type_name"); ok && !strings.Contains(providerSettings.InstanceSizeName, "NVME") {
			providerSettings.DiskTypeName = pointy.Int64(cast.ToInt64(v))
		}
		providerSettings.EncryptEBSVolume = pointy.Bool(true)
	}
	return providerSettings, nil
}
```

**Review by shum — CHANGES_REQUESTED** [review]

**Comment by shum** [comment]
> LGTM but fyi for @shum we didn't actually include the same in https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_advanced_cluster.go so either should work, and advanced is preferred, for those just starting out (we are going to have to move everyone from one to the other). Still good to do till we deprecate clusters.

Good to know. I guess this validation is just to prevent folks who don't read the manual from shooting themselves in the foot.

File: mongodbatlas/resource_mongodbatlas_cluster.go (hunk 1 above)
Comment (martinstibbe) [reply mongodbatlas/resource_mongodbatlas_cluster.go]: @shum I assume you wanted d.Get("provider_disk_type_name") vs pointy.Int64(cast.ToInt64(v)) for the name when it is permitted to set a value ?  i.e.```
 if d.Get("provider_name") == "AZURE" {
		if v, ok := d.GetOk("provider_disk_type_name"); ok && !strings.Contains(providerSettings.InstanceSizeName, "NVME") {
			providerSettings.DiskTypeName = cast.ToString(v)
		}
		providerSettings.EncryptEBSVolume = pointy.Bool(true)
	}
```

**Review by shum — APPROVED** [review]

File: mongodbatlas/resource_mongodbatlas_cluster.go (hunk 1 above)
Comment (shum) [reply mongodbatlas/resource_mongodbatlas_cluster.go]: yea, sorry about that, copy pasted and didn't read everything

**Pull request merged**